### DISCUSSION TITLE
Is there any way to add an action to a ToggleColumn?

### DISCUSSION THREAD

**Discussion by rappasoft** [filamentphp/filament#8545] [General]
I have a ToggleColumn, which works out of the box, but I want to add an action when it is toggled so I can do certain things. But although the IDE autocomplete the action method, it does nothing on the ToggleColumn:

```php
ToggleColumn::make('can_do_thing')
      ->label('Toggle Thing')
      ->action(function (Record $record, string $state): void {
          dd($record); // Never hits
      }),
```

Thanks for any help.

Edit: Answered my own question, in this case use `updateStateUsing` with the same parameters.

**Comment by leandrocfe** [comment]
Yep, you can use `->updateStateUsing()`
